portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -588,7 +588,8 @@ public DLFileEntry copyFileEntry(
 			dlFileEntry.getName());
 
 		DLFileEntry newDLFileEntry = addFileEntry(
-			userId, groupId, repositoryId, destFolderId, sourceFileName,
+			dlFileEntry.getExternalReferenceCode(), userId, groupId,
```

Comment:
@javierdearcos

I think that this is wrong.... I put null on the copy?

Reply:
yep. Should be null to take the externalReferenceCode by default. If not it will throw a DuplicateExternalReferenceCode exception as it should be unique to all file entries